eas-build-archive.md

```diff
@@ -1,10 +1,21 @@
 # How projects are uploaded to EAS Build
 
-When you run `eas build`, we need to get your source code from your development machine to one of our macOS or Linux build workers. In order to do this, EAS CLI will collect and compress project files into a single archive, and then upload that archive to private cloud storage that is accesible only to the build worker. By default, EAS CLI will produce the archive by performing a shallow `git clone` of your project and then compressing the cloned directory with `tar`.
+When you run `eas build`, we need to get your source code from your development machine to one of our macOS or Linux build workers. In order to do this, EAS CLI will collect and compress project files into a single archive, and then upload that archive to private cloud storage that is accessible only to the build worker. By default, EAS CLI will produce the archive by copying all files starting from the root of the git repository with the exception of `.git`, `node_modules` and all files matched by rules from `.gitignore` (or `.easignore` if specified).
 
 ## What files are included in the archive?
 
-You likely do not want to upload your `node_modules` directory, or maybe you have a `.env` file that is ignored by source control. Because EAS CLI uses does shallow `git clone` of your project before compressing it, your `.gitignore` configuration will be leveraged, and only files that are included in the latest commit will be included in the archive.
+You likely do not want to upload your `node_modules` directory, or maybe you have a `.env` file that is ignored by source control. EAS CLI is always respecting `.gitignore` files, but depending on your configuration there might be certain edge cases where eas behavior is not 100% compliant with git.
+
+#### EAS CLI 
+
+By default or if you set `EAS_NO_VCS` environment variable, EAS CLI will use it's own implementation to package the repository, as a result files uploaded to the worker might be a bit different than what is stored in git:
+  - If you have multiple `.gitignore` files, they are applied in isolation starting from the root, so if you have an ignore rule like `test/example` in the parent directory and `!example/example1` in the `test` directory then the entire `example` directory will still be ignored.
+  - The `node_modules` directory is ignored by default.
+  - Even if you are using `git-crypt`, all the files are uploaded as they are in your project directory. This means all sensitive files could be uploaded to EAS Build in a non-encrypted state.
+  - After upload to EAS Build your `.git` directory won't exists, so if you are using any tools that read your vcs information when building that information won't be included (e.g. sentry can read commit hash when uploading sourcemaps).
```

Comment:
```suggestion
  - The `.git` directory is not uploaded to EAS Build. So if you are using any tools that depend on the state of the Git repository, this might result in unexpected behavior (e.g. sentry reads the commit hash when uploading the source maps).
```